__tests__/NCN.test.js

```diff
@@ -0,0 +1,124 @@
+            articleData.forEach((currObject,index) => {expect(typeof currObject).toBe('object')
+            expect(articleData[index]).toEqual(articleDataOrigin[index])})
+        });
+        test('comment data ', () => {
+            commentData.forEach((currObject,index) => {expect(typeof currObject).toBe('object')
+            expect(commentData[index]).toEqual(commentDataOrigin[index])})
+        });
+        test('topic data ', () => {
+            topicData.forEach((currObject,index) => {expect(typeof currObject).toBe('object')
+            expect(topicData[index]).toEqual(topicDataOrigin[index])})
+        });
+        test('user data ', () => {
+            userData.forEach((currObject,index) => {expect(typeof currObject).toBe('object')
+            expect(userData[index]).toEqual(userDataOrigin[index])})
+        });
+    });
+})
+
+//task2
+describe('GET /api/topics',() => {
+    test('GET:200 availble on endpoint /api/topics', () => {
+        return request(app)
+        .get('/api/topics')
+        .expect(200)
+        .then((response) =>  {
+            const body = response.body
+            if(body.length !== 0){
+                body.forEach((dataObj,index) => {
+                    expect(dataObj).toEqual(topicData[index])
+                })
+            }
+        })
+    });
+    test('Get: 404, responds with not found if route does not exist', () => {
+        return request(app)
+        .get('/api/not-a-route')
+        .expect(404)
+        .then((response) => {
+            const body = response.body
+            expect(body.msg).toBe('route does not exist');
+        })
+    });
+})
+
+//task3
+describe('GET /API', () => {
+    test('GET: 200 should return a object of all endpoints', () => {
+        return request(app)
+        .get('/api')
+        .expect(200)
+        .then((response) => {
+            const body = response.text
+            expect(typeof response).toBe('object')
+            expect(JSON.parse(body)).toEqual(endpoints)
+        })
+    });
+})
+
+//task4
+describe('GET api/article/:articleid', () => {
+    test('GET: 200 should resturn an object', () => {
+        return request(app)
+        .get('/api/articles/1')
+        .expect(200)
+        .then((response) => {
+            expect(typeof response).toBe('object')
+        })
+    });
+    test('GET:200 should expected properties ', () => {
+        return request(app)
+        .get('/api/articles/1')
+        .expect(200)
+        .then((response) => {
+            const body = response.body
+            expect(body).toHaveProperty('author')
+            expect(body).toHaveProperty('title')
+            expect(body).toHaveProperty('article_id')
+            expect(body).toHaveProperty('body')
+            expect(body).toHaveProperty('topic')
+            expect(body).toHaveProperty('created_at')
+            expect(body).toHaveProperty('votes')
+            expect(body).toHaveProperty('article_img_url')
```

Comment:
This works fine but going forward look into jest matchers expect.objectContaining or .toMatchObject they allow for more robust and flexible testing.